### ISSUE TITLE
Newly added node runs out of space

### ISSUE THREAD

**Issue by KnifeyMoloko** [scylladb/scylla#10549]
## Installation details

Kernel Version: 5.13.0-1022-aws
Scylla version (or git commit hash): `5.1.dev-20220504.b26a3da584cc` with build-id `ab2a33a30756c1513f4c516cd272291e75acec0e`
Cluster size: 3 nodes (i3.4xlarge)

Scylla Nodes used in this run:
- gemini-1tb-10h-master-oracle-db-node-53b60c69-1 (54.246.72.73 | 10.0.0.221) (shards: 14)
- gemini-1tb-10h-master-db-node-53b60c69-4 (52.19.45.163 | 10.0.0.44) (shards: 14)
- gemini-1tb-10h-master-db-node-53b60c69-3 (3.251.79.228 | 10.0.1.225) (shards: 14)
- gemini-1tb-10h-master-db-node-53b60c69-2 (54.229.71.139 | 10.0.0.64) (shards: 14)
- gemini-1tb-10h-master-db-node-53b60c69-1 (34.248.185.187 | 10.0.2.112) (shards: 14)

OS / Image: `ami-0961efebe1f65ab2e` (aws: eu-west-1)

Test: `gemini-1tb-10h`
Test id: `53b60c69-b154-41db-b95d-ff2b50dd8e98`
Test name: `scylla-master/gemini-/gemini-1tb-10h`
Test config file(s):

- [gemini-1tb-10h.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/ea4aaa1b4f8b631fea2df71d8206e815c3d7614d/test-cases/gemini/gemini-1tb-10h.yaml)

## Issue description

**>>>>>>>**
During this run we:

1. Decommissioned a node and added a new one to the cluster (node 4).
```
[DecommissionStreamingErr - adding a new node]
05:23:39 - Bootstrap starts 
05:54:25 - Bootstrap succeeded
05:54:26 - node state jump to normal
05:55:19 - Rebuild starts
06:14:53 - Rebuild succeeded
```
2. After the initial Rebuild process, we killed the added node.
```
[KillScylla - kills node 4]
May 06 06:20:21 gemini-1tb-10h-master-db-node-53b60c69-4 systemd[1]: scylla-jmx.service: Main process exited, code=killed, status=9/KILL
May 06 06:20:21 gemini-1tb-10h-master-db-node-53b60c69-4 systemd[1]: scylla-jmx.service: Failed with result 'signal'.
May 06 06:20:21 gemini-1tb-10h-master-db-node-53b60c69-4 systemd[1]: scylla-jmx.service: Scheduled restart job, restart counter is at 1.
...
<Restart>
<Lots of removing temporary sstable directoies after restart>
...
May 06 06:21:32 gemini-1tb-10h-master-db-node-53b60c69-4 scylla[12532]:  [shard 0] sstable_directory - Table ks1.table1 with compaction strategy SizeTieredCompactionStrategy found SSTables that need reshape. Starting reshape process
...
May 06 06:37:34 gemini-1tb-10h-master-db-node-53b60c69-4 scylla[12532]:  [shard 0] database - Reshaped 893GB in 961.53 seconds, 929MB/s
...
May 06 06:38:48 gemini-1tb-10h-master-db-node-53b60c69-4 scylla[12532]:  [shard 0] storage_service - Node 10.0.0.64 state jump to normal
...
```
3. Altered a table using: `ALTER TABLE ks1.table      1 WITH caching = {'keys': 'NONE', 'rows_per_partition': 'NONE'};`
```
[ModifyTable - ALTER TABLE ks1.table1 WITH caching = {'keys': 'NONE', 'rows_per_partition': 'NONE'};]
May 06 06:57:52 gemini-1tb-10h-master-db-node-53b60c69-4 scylla[12532]:  [shard 1] schema_tables - Altering ks1.table1 id=75c25bf0-cced-11ec-a177-720a26afc6ae version=e7838fb4-53d0-33d7-81c7-6ba32507642e
May 06 06:57:52 gemini-1tb-10h-master-db-node-53b60c69-4 scylla[12532]:  [shard 1] query_processor - Column definitions for ks1.table1 changed, invalidating related prepared statements
...
<Compactions>
...
```
4. Destroyed some of the data (on node 4) and restarted Scylla.
```
[DestroyDataThenRebuild]
May 06 07:14:38 gemini-1tb-10h-master-db-node-53b60c69-4 systemd[1]: Stopping Scylla JMX...
May 06 07:14:38 gemini-1tb-10h-master-db-node-53b60c69-4 systemd[1]: scylla-jmx.service: Main process exited, code=exited, status=143/n/a
...
...
May 06 07:15:57 gemini-1tb-10h-master-db-node-53b60c69-4 scylla[18169]:  [shard 0] init - Scylla version 5.1.dev-0.20220504.b26a3da584cc with build-id ab2a33a30756c1513f4c516cd272291e75acec0e starting ...
May 06 07:16:00 gemini-1tb-10h-master-db-node-53b60c69-4 systemd[1]: Started Scylla Server.
...
May 06 07:16:01 gemini-1tb-10h-master-db-node-53b60c69-4 scylla[18169]:  [shard 0] schema_tables - Schema version changed to 903817e8-25cc-383e-a82b-bd8c0fcebd89
...
May 06 07:17:03 gemini-1tb-10h-master-db-node-53b60c69-4 scylla[18169]:  [shard 0] range_streamer - Rebuild starts, nr_ranges_remaining=3569
```
5. After these 2 restarts and 1 table alteration the node reported `No space left on device`:
```
022-05-06T08:08:14+00:00 gemini-1tb-10h-master-db-node-53b60c69-4 !     ERR |  [shard 11] storage_service - Shutting down communications due to I/O errors until operator intervention: Disk error: std::system_error (error system:28, No space left on device)
```

Throughout the process node 4 was using more and more of its disk space:

![Screenshot from 2022-05-11 18-21-47](https://user-images.githubusercontent.com/13716678/167925107-eb451e24-4b05-4879-bc85-8ec78cd08dea.png)

**<<<<<<<**

- Restore Monitor Stack command: `$ hydra investigate show-monitor 53b60c69-b154-41db-b95d-ff2b50dd8e98`
- Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?test_id=53b60c69-b154-41db-b95d-ff2b50dd8e98)
- Show all stored logs command: `$ hydra investigate show-logs 53b60c69-b154-41db-b95d-ff2b50dd8e98`

## Logs:
- **db-cluster-53b60c69.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/53b60c69-b154-41db-b95d-ff2b50dd8e98/20220506_082527/db-cluster-53b60c69.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/53b60c69-b154-41db-b95d-ff2b50dd8e98/20220506_082527/db-cluster-53b60c69.tar.gz)
- **monitor-set-53b60c69.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/53b60c69-b154-41db-b95d-ff2b50dd8e98/20220506_082527/monitor-set-53b60c69.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/53b60c69-b154-41db-b95d-ff2b50dd8e98/20220506_082527/monitor-set-53b60c69.tar.gz)
- **loader-set-53b60c69.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/53b60c69-b154-41db-b95d-ff2b50dd8e98/20220506_082527/loader-set-53b60c69.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/53b60c69-b154-41db-b95d-ff2b50dd8e98/20220506_082527/loader-set-53b60c69.tar.gz)
- **sct-runner-53b60c69.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/53b60c69-b154-41db-b95d-ff2b50dd8e98/20220506_082527/sct-runner-53b60c69.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/53b60c69-b154-41db-b95d-ff2b50dd8e98/20220506_082527/sct-runner-53b60c69.tar.gz)

[Jenkins job URL](https://jenkins.scylladb.com/job/scylla-master/job/gemini-/job/gemini-1tb-10h/156/)

**Comment by roydahan** [comment]
There is something we need to check here.
The rebuild at 05:55 is not expected.

**Comment by KnifeyMoloko** [comment]
@roydahan That was also my initial thought, but it turns out to be expected. The `DecommissionStreaminErr` that was running then uses the method `break_streaming_task_and_rebuild` which does what the name says. It triggered the rebuild.

**Comment by enaydanov** [comment]
Got pretty similar thing on 2022.1:


## Installation details

Kernel Version: 5.13.0-1021-aws
Scylla version (or git commit hash): `2022.1~rc3-20220406.5cc3b678c` with build-id `48dfae0735cd8efc4ae2f5c777beaee2a1e89f4a`
Cluster size: 3 nodes (i3.4xlarge)

Scylla Nodes used in this run:
- gemini-1tb-10h-2022-1-oracle-db-node-be6ad0cf-1 (34.242.135.214 | 10.0.2.94) (shards: 14)
- gemini-1tb-10h-2022-1-db-node-be6ad0cf-4 (3.248.193.196 | 10.0.1.96) (shards: 14)
- gemini-1tb-10h-2022-1-db-node-be6ad0cf-3 (54.246.19.55 | 10.0.1.94) (shards: 14)
- gemini-1tb-10h-2022-1-db-node-be6ad0cf-2 (34.248.210.204 | 10.0.0.248) (shards: 14)
- gemini-1tb-10h-2022-1-db-node-be6ad0cf-1 (34.242.222.137 | 10.0.1.252) (shards: 14)


OS / Image: `ami-071c70d20f0fdbb2c` (aws: eu-west-1)

Test: `gemini-1tb-10h-test`
Test id: `be6ad0cf-e6d8-4957-bcb2-16db4a45bf0c`
Test name: `enterprise-2022.1/gemini-/gemini-1tb-10h-test`
Test config file(s):

- [gemini-1tb-10h.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/dd48c8d642e27935eb2d231329c34ee737683ab9/test-cases/gemini/gemini-1tb-10h.yaml)

## Issue description

![image](https://user-images.githubusercontent.com/63434/170692005-224fc48b-5610-4706-a3c6-45c14923f354.png)

- Restore Monitor Stack command: `$ hydra investigate show-monitor be6ad0cf-e6d8-4957-bcb2-16db4a45bf0c`
- Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?test_id=be6ad0cf-e6d8-4957-bcb2-16db4a45bf0c)
- Show all stored logs command: `$ hydra investigate show-logs be6ad0cf-e6d8-4957-bcb2-16db4a45bf0c`

## Logs:
- **db-cluster-be6ad0cf.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/be6ad0cf-e6d8-4957-bcb2-16db4a45bf0c/20220504_222632/db-cluster-be6ad0cf.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/be6ad0cf-e6d8-4957-bcb2-16db4a45bf0c/20220504_222632/db-cluster-be6ad0cf.tar.gz)
- **monitor-set-be6ad0cf.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/be6ad0cf-e6d8-4957-bcb2-16db4a45bf0c/20220504_222632/monitor-set-be6ad0cf.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/be6ad0cf-e6d8-4957-bcb2-16db4a45bf0c/20220504_222632/monitor-set-be6ad0cf.tar.gz)
- **loader-set-be6ad0cf.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/be6ad0cf-e6d8-4957-bcb2-16db4a45bf0c/20220504_222632/loader-set-be6ad0cf.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/be6ad0cf-e6d8-4957-bcb2-16db4a45bf0c/20220504_222632/loader-set-be6ad0cf.tar.gz)
- **sct-runner-be6ad0cf.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/be6ad0cf-e6d8-4957-bcb2-16db4a45bf0c/20220504_222632/sct-runner-be6ad0cf.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/be6ad0cf-e6d8-4957-bcb2-16db4a45bf0c/20220504_222632/sct-runner-be6ad0cf.tar.gz)


[Jenkins job URL](https://jenkins.scylladb.com/job/enterprise-2022.1/job/gemini-/job/gemini-1tb-10h-test/3/)

**Comment by roydahan** [comment]
@enaydanov this is probably because we (you) haven't backport the fix above from @KnifeyMoloko to 2022.1.
Please check if it still happens in master after the fix.

**Comment by enaydanov** [comment]
> @enaydanov this is probably because we (you) haven't backport the fix above from @KnifeyMoloko to 2022.1. Please check if it still happens in master after the fix.

Yep, my bad.
